### Задача 1
Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):  
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],  
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].  
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy  
Полученные значения должны быть равны.  
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,  
а затем с использованием функций из библиотек numpy и pandas. 


**Решение:**  
Найдем величину ковариации по формуле:
$$ cov_{xy}=M(x*y) - M(x)*M(y)$$

In [29]:
import numpy as np
import scipy.stats as stats

In [30]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])  
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

cov_manual = np.mean(zp*ks) - np.mean(zp)*np.mean(ks)
cov_manual

9157.839999999997

In [31]:
cov_auto = np.cov(zp,ks,ddof=0)
#меняем ddof , так как по формуле вручную считаем смещенную величину(нет корректировки числа наблюдений)
cov_auto

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

Получили **величину ковариации** между зарплатой и скорингом в 9157.84  
Теперь найдем коэффициент корреляции Пирсона, вручную по формуле:
$$ r_{xy}=\frac{cov_{xy}}{\sigma_x * \sigma_y} $$

In [32]:
pirs_manual = cov_manual / (np.std(zp,ddof=0) * np.std(ks,ddof=0))
pirs_manual

0.8874900920739158

In [33]:
#теперь через встроенную функцию Numpy
np.corrcoef(zp,ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [34]:
#и через Pandas, как просили
import pandas as pd

p_zp= pd.Series(zp)
p_ks= pd.Series(ks)
p_zp.corr(p_ks , method='pearson')

0.8874900920739162

Значения совпали, **коэффициент корреляции Пирсона** равен 0.887

### Задача 2
Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:  
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.  
Известно, что в генеральной совокупности IQ распределен нормально.  
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

*Решение:*  
Так как известно о нормальности распределения, но неизвестна дисперсия ГС(генеральной совокупности), то для вычисления доверительного интервала будем использовать критическое значение криттерия Стьюдента, и среднее квадратичное отклонение выборки:

In [35]:
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
#отклонение
iq_sigma = np.std(iq, ddof=1)
iq_sigma



10.54566788359614

Чтобы найти табличное значение крит. Стьюдента при надежности 0.95 , мы должен из полной вероятности 1 вычесть $\alpha/2 = 0.025$  , так как это двусторонняя отсечка статистической значимости. 

In [36]:
#табличное значение крит. Стьюдента при надежности 0.95  - 
t_05 = stats.t.ppf(1-(1-0.95)/2, len(iq)-1)
t_05

2.2621571627409915

Осталось найти интервалы по формуле:
$$ \bar{X} \pm t_{\frac{\alpha}{2}}*\frac{\sigma_x}{\sqrt(n)} $$

In [37]:
np.mean(iq)

118.1

In [38]:
print("Доверительные интервал c надежностью в 95% : \n")
print("[{bor1:.3f};{bor2:.3f}]".format(bor1 = np.mean(iq) - t_05*iq_sigma/np.sqrt(len(iq)) ,
                                        bor2 =np.mean(iq) + t_05*iq_sigma/np.sqrt(len(iq))))


Доверительные интервал c надежностью в 95% : 

[110.556;125.644]


### Задача 3
Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см.  Объем выборки равен 27,
среднее выборочное составляет 174.2.  
Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

*Решение:*
Дано:  
$D(X) = 25$ кв.см. , отсюда $=> \sigma = \sqrt(25) = 5$ см  
$n = 27$  
$M(X) = 174.2$  
Так как известны математические параметры ГС, то будем использовать Z-криттерий и формулу интервала:
$$ M(X) \pm Z_{\frac{\alpha}{2}}*\frac{\sigma}{\sqrt(n)} $$
$$ 174.2 \pm 1.96*\frac{5}{\sqrt(27)} $$
$$ 174.2 \pm 1.886 $$
Получим интервал в  [172.314;176.086]


Проверим через расчет в питоне:

In [39]:

z_par= stats.norm.ppf(1-0.05/2)
z_par

1.959963984540054

In [40]:
n=27
sigma = 5
central_tendency = 174.2
print("Доверительные интервал c надежностью в 95% : \n")
print("[{bor1:.3f};{bor2:.3f}]".format(bor1 = central_tendency - z_par*sigma/np.sqrt(n) ,
                                        bor2 =central_tendency + z_par*sigma/np.sqrt(n)))

Доверительные интервал c надежностью в 95% : 

[172.314;176.086]
